tests/nested/core/hotplug/task.yaml

```diff
@@ -0,0 +1,185 @@
+summary: Create ubuntu core image, install snapd and test hotplug feature
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    create_nested_core_vm
+
+    snap pack "$TESTSLIB"/snaps/serial-port-hotplug
+    copy_remote serial-port-hotplug_1.0_all.snap
+
+    if is_core_18_nested_system; then
+        execute_remote "sudo snap install --devmode jq-core18"
+        execute_remote "sudo snap alias jq-core18.jq jq"
+    else
+        execute_remote "sudo snap install --devmode jq"
+    fi
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+    destroy_nested_vm
+    rm -f /tmp/serialport{0,1}
```

Comment:
Nitpick, this can go away (my oversight when I gave you the diff) as we don't have dialout user nor write to the virtual port here in core test. It's fine to leave it for now and clean up in followup.